code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -165,7 +179,7 @@
 			icon_beaker = image('icons/obj/chemical.dmi', src, "disp_beaker") //randomize beaker overlay position.
 		icon_beaker.pixel_x = rand(-10,5)
 		overlays += icon_beaker
-	else if(user.a_intent != "harm")
+	else if(user.a_intent != "harm" && !istype(I, /obj/item/weapon/card/emag))
 		user << "<span class='warning'>You can't load \the [I] into the machine!</span>"
```

Comment:
this check on line 182 might be kinda snowflake but it works. if you have a better solution please tell me